joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/passes/ImportResolverPass.scala

```diff
@@ -35,9 +39,37 @@ class ImportResolverPass(cpg: Cpg, packageTableInfo: PackageTable) extends XImpo
      This needs to be handled accordingly
      */
 
-    methodTableModelList.flatMap { methodTableModel =>
-      Seq(ResolvedMethod(s"$expEntity::program:${methodTableModel.parentClassPath.stripSuffix(":")}.new", "new"))
-    }.toSet
+    val relativeImportNodes = if (importCall.name.equals("require_relative")) {
+      val rawEntity   = expEntity.stripPrefix("./")
+      val matcher     = pathPattern.matcher(rawEntity)
+      val sep         = Matcher.quoteReplacement(JFile.separator)
+      val root        = codeRoot + JFile.separator
+      val currentFile = s"$root$fileName"
+      val entity      = if (matcher.find()) matcher.group(1) else rawEntity
+      val resolvedPath = better.files
+        .File(currentFile.stripSuffix(currentFile.split(sep).last), entity.split("\\.").head)
```

Comment:
`head` and `last` are pretty unsafe, could try implementing this with `headOption` and `lastOption`?

Reply:
Ok, will make the change.